code/gateway/dataaccess.q

```diff
@@ -171,27 +173,45 @@ partdict:{[input]
 adjustqueries:{[options;part]
     // if only one process then no need to adjust
     if[2>count p:options`procs;:options];
-    // get the date casting where relevant
-    st:$[a:-14h~tp:type start:options`starttime;start;`date$start];
-    et:$[a;options`endtime;`date$options`endtime];
-    // get the dates that are required by each process
-    dates:group key[part]where each{within[y;]each value x}[part]'[l:st+til 1+et-st];
-    dates:l{(min x;max x)}'[dates];
-    // if start/end time not a date, then adjust dates parameter for the
-    // correct types
-    if[not a;
-        // converts dates dictionary to timestamps/datetimes
-        dates:$[-15h~tp;{"z"$x};::]{(0D+x 0;x[1]+1D-1)}'[dates];
+    // get the tablename
+    tabname:options[`tablename];
+    // remove duplicate servertypes from the gw.servers
+    servers:select from .gw.servers where i=(first;i)fby servertype;
+    // extract the procs which have the table defined
+    servers:select from servers where {[x;tabname]tabname in @[x;`tables]}[;tabname] each attributes;
+    // create a dictionary of the attributes against servertypes
+    procdict:(exec servertype from servers)!(exec attributes from servers)@'(key each exec attributes from servers)[;0];
+    // if the response is a dictionary index into the tablename
+    procdict:@[procdict;key procdict;{[x;tabname]if[99h=type x;:x[tabname]];:x}[;tabname]];
+    // create list of all available partitions
+    possParts:raze value procdict;
+
+    //group partitions to relevant process
+    partitions:group key[part]where each{within[y;]each value x}[part]'[possParts];
+    partitions:possParts{(min x;max x)}'[partitions];
+    partitions:`timestamp$partitions;
+
+    // adjust the times to account for period end time when int partitioned
+    c:first[x:@[partitions;`hdb]],-1+ first[@[partitions;`rdb]];
+    d:first[@[partitions;`rdb]],options `endtime;
```

Comment:
```suggestion
    c:first[x:partitions`hdb],-1+ first[partitions`rdb];
    d:first[partitions`rdb],options `endtime;
```

`@` isn't needed here, IMO clearer without

does `x` get used again? if not, don't set it here